src/Psalm/Internal/CallMap.php

```diff
@@ -393,7 +393,7 @@
 'array_rand' => ['int|string|array<int,int>|array<int,string>', 'input'=>'array', 'num_req'=>'int'],
 'array_rand\'1' => ['int|string', 'input'=>'array'],
 'array_reduce' => ['mixed', 'input'=>'array', 'callback'=>'callable(mixed,mixed):mixed', 'initial='=>'mixed'],
-'array_replace' => ['array', 'arr1'=>'array', 'arr2'=>'array', '...args='=>'array'],
+'array_replace' => ['?array', 'arr1'=>'array', 'arr2'=>'array', '...args='=>'array'],
```

Comment:
`array_replace` returns null for non-array types, which Psalm checks: https://github.com/php/php-src/blob/master/ext/standard/array.c#L3746-L3752